### PR TITLE
Hw06 Nikita Dukin

### PR DESCRIPTION
**Pull request by Fant1k34**

### CODE REVIEW

File: src/IO.hs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+getLineHelper container = getChar >>= \r -> return (container ++ [r])
+
+-- Implement reading line from standard input. 
+-- Use getChar to read a single character. 
+myGetLine' :: [Char] -> IO String
+myGetLine' state = do
+    array <- getLineHelper state
+    if last array == '\n'
+        then return array
+    else getLineHelper array >>= myGetLine'
```
Comment (kajigor) [inline src/IO.hs:14]: `myGetLine'` is not supposed to have an additional parameter.

File: src/IO.hs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+getLineHelper :: [Char] -> IO [Char]
+getLineHelper container = getChar >>= \r -> return (container ++ [r])
+
+-- Implement reading line from standard input. 
+-- Use getChar to read a single character. 
+myGetLine' :: [Char] -> IO String
+myGetLine' state = do
+    array <- getLineHelper state
+    if last array == '\n'
```
Comment (kajigor) [inline src/IO.hs:12]: Do you really need to inspect the array for the last character?

File: src/IO.hs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    putStrLn ("Hello, " ++ name)
+
+
+-- Use interact in helloUser.
+helloUser' :: IO ()
+helloUser' = do
+    putStrLn "Я не до конца понял, что должна делать эта функция"
+    putStrLn "Но вот сейчас я взаимодействую вместе с helloUser"
+    putStrLn "Сейчас я поприветсвую по имени"
+    helloUser
```
Comment (kajigor) [inline src/IO.hs:33]: `interact :: (String -> String) -> IO ()` reads the line from the standard input, applies the input function to it, and outputs the result to the standard output.

**Review by kajigor — CHANGES_REQUESTED** [review]

**Comment by kajigor** [comment]
1 ok

File: src/IO.hs (hunk 3 above)
Comment (Fant1k34) [reply src/IO.hs]: Thank you, I've used this method in the implementation helloUser'

File: src/IO.hs (hunk 2 above)
Comment (Fant1k34) [reply src/IO.hs:12]: Yes... I tried to delete this if-condition, but it's not working (endless IO operation unless ctlr + C is pressed)

**Pull request closed without merging**